SimTKmath/Integrators/src/CPodes/sundials/src/cpodes/cpodes.c

```diff
@@ -1615,7 +1615,7 @@ void CPodeFree(void **cpode_mem)
 
   cp_mem = (CPodeMem) (*cpode_mem);
   
-  cpFreeVectors(cp_mem);
+  if (cp_mem->cp_MallocDone) cpFreeVectors(cp_mem);
```

Comment:
> If your proposed fix is the correct one, I wonder why CVODES doesn't have similar logic. 

I looked at that too, and I do believe this code would also cause the same problem. I think, however, this is intentional as the comment in this version and the one used by Simbody does say that it frees memory by `CPodeInit()`. However, Simbody does not call `CPodeInit()` until initialization, while Simbody (in a destructor) does call `CPodeFree()` regardless if `CPodeInit()` has or has not been called. 

Another way to go about this would be to use the `initialized` flag on the Simbody level to control the line here: https://github.com/simbody/simbody/blob/master/SimTKmath/Integrators/src/CPodesIntegrator.cpp#L158. However, in this case, I'm still not sure who will delete `cp_mem` (in either version of the code). At least in my digging, it seems like the code assumes that `CPodeInit()` should always be called after `CPodeCreate()`, and if it isn't, then there will be a leak regardless?

> Consider using FALSE (if (cp_mem->cp_MallocDone == FALSE)) for consistency with the rest of the file.

I was trying to be consistent with the lines just below, L1628 and 1630. Which lines were you thinking of?

Reply:
Ah I think I understand. LGTM.

> I was trying to be consistent with the lines just below, L1628 and 1630. Which lines were you thinking of?

Ah I see; nevermind. I was looking at https://github.com/simbody/simbody/blob/5d48fcacf214a54ad8cbe105db16dae54a81ce00/SimTKmath/Integrators/src/CPodes/sundials/src/cpodes/cpodes.c#L534